sympy/assumptions/handlers/order.py

```diff
@@ -66,32 +66,88 @@ def Add(expr, assumptions):
         if r is not True:
             return r
 
-        nonpos = 0
-        for arg in expr.args:
-            if ask(Q.negative(arg), assumptions) is not True:
-                if ask(Q.positive(arg), assumptions) is False:
-                    nonpos += 1
-                else:
-                    break
-        else:
-            if nonpos < len(expr.args):
-                return True
+
+        args = [arg for arg in expr.args if not ask(Q.zero(arg), assumptions)]
+        if not args:
+            return False
+
+        inf = neg_inf = pos_inf = neg = npos = nneg = False
+
+        for arg in args:
+            if pos_inf and neg_inf:
+                return None
+
+            _inf = ask(Q.infinite(arg), assumptions)
+            if _inf:
+                inf = True
+
+            n = ask(Q.negative(arg), assumptions)
+            if n:
+                if _inf is not False:
+                    neg_inf = True
+                neg = True
+                continue
+
+            np = ask(Q.nonpositive(arg), assumptions)
+            if np:
+                if _inf is True:
+                    neg_inf = True
+                npos = True
+                continue
+
+            nn = ask(Q.nonnegative(arg), assumptions)
+            if nn:
+                if _inf or (_inf is None and nn is False):
+                    pos_inf = True
+                nneg = True
+                continue
+            else:
+                return None
+        if pos_inf and neg_inf:
+            return None
+        elif inf and pos_inf:
+            return False
+        elif inf and neg_inf:
+            return True
```

Comment:
If the result is negative infinity then it should not return True for negative because negative implies finite. Negative infinity is `extended_negative`.

Reply:
This code is all probably unreachable anyway, because this function starts with:
```
        r = ask(Q.real(expr), assumptions)
        if r is not True:
            return r
```
So by the time we get here, we know there are no infinities.

Reply:
I commented this above in https://github.com/sympy/sympy/pull/19241#discussion_r426274523, but it looks like it was marked resolved by someone.

Reply:
Overlooked this. I have changed it now.